app/views/govuk_publishing_components/components/docs/contextual_guidance.yml

```diff
@@ -22,8 +22,20 @@ examples:
       guidance_id: news-title-guidance
       content: |
         <p>The title must make clear what the content offers users. Use the words your users do to help them find this. Avoid wordplay or teases.</p>
-      block: |
-        <div class="govuk-form-group">
-          <label class="gem-c-label govuk-label" for="news-title">News title</label>
-          <input id="news-title" type="text" class="gem-c-input govuk-input" aria-describedby="news-title-guidance">
-        </div>
+    embed: |
```

Comment:
I think this could be simplified as follows, using the `block` option. I also think highlighting the bit that is the optional, example content is worth doing, to help developers when copying and pasting these examples.

```
examples:
  default:
    description: Reveals a contextual guidance on the side overflowing the container
    data:
      html_for: news-title
      title: Writing a news title
      guidance_id: news-title-guidance
      content: |
        <p>The title must make clear what the content offers users. Use the words your users do to help them find this. Avoid wordplay or teases.</p>
      block: |
        <!-- example content -->
          <%= render "govuk_publishing_components/components/input", { 
              label: {
                text: "News title"
              },
              name: "news-title",
              id: "news-title",
              describedby: "news-title-guidance"
            } 
          %>
        <!-- end of example content -->
```

This gives:

![Screenshot 2023-05-31 at 13 35 19](https://github.com/alphagov/govuk_publishing_components/assets/861310/1d335220-b9c7-42a5-a797-826272dde628)

Reply:
I think this technique could be applied to the other examples in this commit. Happy to discuss though.

Reply:
Thanks for this, looks much better now. The only components that I could not update to use this approach are `modal-dialogue` and `textarea`. For modal-dialogue, I used embed to include the `button` component that launches the modal. For the textarea contextual guidance example, I needed to embed the textarea component within the contextual-guidance component.